WooCommerce/Classes/Analytics/WooAnalytics.swift

```diff
@@ -98,16 +90,52 @@ public extension WooAnalytics {
     ///   - error: the error to track
     ///
     func track(_ stat: WooAnalyticsStat, withError error: Error) {
+        track(stat, properties: nil, withError: error)
+    }
+
+    /// Track a specific event with associated properties and an associated error (that is translated to properties)
+    ///
+    /// - Parameters:
+    ///   - stat: the event name
+    ///   - properties: a collection of properties related to the event
+    ///   - error: the error to track
+    ///
+    func track(_ stat: WooAnalyticsStat, properties passedProperties: [AnyHashable: Any]?, withError error: Error?) {
         guard userHasOptedIn == true else {
             return
         }
 
+        let properties = combinedProperties(from: error, with: passedProperties)
+
+        if let updatedProperties = updatePropertiesIfNeeded(for: stat, properties: properties) {
+            analyticsProvider.track(stat.rawValue, withProperties: updatedProperties)
+        } else {
+            analyticsProvider.track(stat.rawValue)
+        }
+    }
+
+    private func combinedProperties(from error: Error?, with passedProperties: [AnyHashable: Any]?) -> [AnyHashable: Any]? {
+        let properties: [AnyHashable: Any]?
+        let errorProperties = errorProperties(from: error)
+
+        if let passedProperties = passedProperties {
+            properties = passedProperties.merging(errorProperties ?? [:], uniquingKeysWith: { current, _ in
+                current
+            })
+        } else {
+            properties = errorProperties
+        }
+        return properties
```

Comment:
nit: non-blocking suggestion

```suggestion
        let passedProperties: [AnyHashable: Any] = passedProperties ?? [:]
        let errorProperties = errorProperties(from: error) ?? [:]
        return passedProperties.merging(errorProperties, uniquingKeysWith: { current, _ in
            current
        })
```

Reply:
Unfortunately, nil-coalescing an empty dictionary for `passedProperties` will result in a change in behaviour, which causes [tests to fail](https://github.com/woocommerce/woocommerce-ios/blob/4fb64991809c42bdd99e5edad136062d0cb8ea8d/WooCommerce/WooCommerceTests/System/WooAnalyticsTests.swift#L28) by passing an empty dictionary where previously it would have returned nil. This [affects four events](https://github.com/woocommerce/woocommerce-ios/blob/4fb64991809c42bdd99e5edad136062d0cb8ea8d/WooCommerce/Classes/Analytics/WooAnalyticsStat.swift#L545).

I approached it the same as you have here initially, but wanted to preserve the existing behaviour of the two existing functions while I combined the code into a single function.

I think it would be good to change to this more readable approach, and since we're passing an empty dictionary around it shouldn't have any impact on what gets sent to Tracks, so I'll do that in a subsequent PR.